docs(features): complete assets page (#4563)
nuxt/framework: docs/content/2.guide/2.features/3.assets.md

# Assets

Nuxt uses two directories to handle assets like stylesheets, fonts or images:

> - The `assets/` directory contains every asset that you want the build tool (Vite or Webpack) to process.

Reviewer: Can we link to the assets directory here?

> - The [`assets/` directory](/guide/directory-structure/assets) contains every asset that you want the build tool (Vite or Webpack) to process.

Reviewer 2: (the fact is, `assets/` is mostly convention for where to put assets not an auto scanned dir by nuxt. i wish we could little bit make it clear that other directories can be used)

Author: Thinking about this, I think we might switch the two sections, starting with `public/` : 
1. `public/` directory, the static server
2. `assets/` directory, the **convention** for files that you want processed and not served as-is.

what do you think?

> - The `public/` directory content is served at the server root as-is.

Reviewer: Can we link to the public dir as well?

## `assets/` directory

> Nuxt uses Vite or Webpack to build and bundle your application. The main function of these build tools is to process JavaScript files, but they can be extended through [plugins](https://vitejs.dev/plugins/) (for Vite) or [loaders](https://webpack.js.org/loaders/) (for Webpack) to process other kind of assets, like stylesheets, fonts or SVG. This step transforms the original file mainly for performance or caching purposes (such as stylesheets minification or browser cache invalidation).

Reviewer 2: ```suggestion
Nuxt uses [Vite](https://vitejs.dev/guide/assets.html) or [Webpack](https://webpack.js.org/guides/asset-management/) to build and bundle your application. The main function of these build tools is to process JavaScript files, but they can be extended through [plugins](https://vitejs.dev/plugins/) (for Vite) or [loaders](https://webpack.js.org/loaders/) (for Webpack) to process other kind of assets, like stylesheets, fonts or SVG. This step transforms the original file mainly for performance or caching purposes (such as stylesheets minification or browser cache invalidation).
```

> In your application's code, you can reference a file located in the `assets/` directory by using the `~/assets/` path or `assets/` alias.

Reviewer 2: Let's not advocate for ambiguous route (`assets/`)

```suggestion
In your application's code, you can reference a file located in the `assets/` directory by using the `~/assets/` path.
```

::alert{type=info icon=💡}
Nuxt won't serve files in the `assets/` directory at a static URL like `/assets/my-file.png`. If you need a static URL, use the [`public/` directory](#public-directory).
::

### Example

For example, referencing an image file that will be processed if a build tool is configured to handle this file extension:

```vue [app.vue]
<template>
  <img src="assets/img/nuxt.png" alt="Discover Nuxt 3" />
</template>
```

> ## `public/` directory

Reviewer 2: I would move this section above assets. Seems more easier to understand and then we can continue with assets and why bundling and hashing is sometimes better

Author: Great point, so much easier indeed.

> The `public/` directory is used as a public server for every asset that build tools don't need to process. This includes files where names can't be changed, for example `robots.txt`, or that you want to be publicly available at a defined URL of your application.

Reviewer 3: ```suggestion
The `public/` directory is used as a public server for static assets that build tools don't need to process. This includes files whose names cannot be changed (for example `robots.txt`), or the ones you want to be publicly available at a defined URL of your application.
```

> The `public/` directory is used as a public server for static assets that build tools don't need to process. This includes files whose names cannot be changed (for example `robots.txt`), or the ones you want to be publicly available at a defined URL of your application.

Reviewer 2: (we can link to directory structure section)

You can get a file in the `public/` directory from your application's code or from a browser by the root URL `/`.

::alert{type=info}

> Nuxt can be configured to use a different name for this directory in the [`nuxt.config` file](/api/configuration/nuxt.config#public)

Reviewer 3: ```suggestion
Nuxt can be configured to use a different name for this directory in the [`nuxt.config` file](/api/configuration/nuxt.config#public).
```
